### PR TITLE
Docs: Natspec for services folder

### PR DESCRIPTION
**Pull request by annipi**
NatSpec documentation for:

- IService
- Service
- ILendingService
- LendingService
- IBorrowService
- BorrowService
- ServiceTypeManager

### CODE REVIEW

File: contracts/services/IBorrowService.sol [hunk 1]
```diff
@@ -3,7 +3,19 @@ pragma solidity ^0.8.16;
...
+ */
 interface IBorrowService {
+    /**
+     * @notice Emitted when a new borrow service is consumed
+     * @param listingId The id of the listing
+     * @param borrower The address of the borrower
+     * @param currency The address of the currency
+     * @param amount The amount of the loan
+     */
     event Borrow(
```
Comment (davidfragalaureano) [inline contracts/services/IBorrowService.sol:19]: Duration is missing in the NatSpec comment.

**Review by davidfragalaureano — APPROVED** [review]
LG

File: contracts/services/IBorrowService.sol (hunk 1 above)
Comment (annipi) [reply contracts/services/IBorrowService.sol:19]: good catch

File: contracts/services/IBorrowService.sol [hunk 2]
```diff
@@ -12,13 +25,28 @@ interface IBorrowService {
...
+     */
     event Pay(
         uint256 indexed listingId,
         address indexed borrower,
         address indexed currency,
         uint256 amount
     );
 
+    /**
+     * @notice Borrows funds to an user using from a service listing
```
Comment (dvrosalesm) [inline contracts/services/IBorrowService.sol:43]:
```suggestion
     * @notice Borrows funds to a user from a service listing
```

File: contracts/services/IBorrowService.sol [hunk 3]
```diff
@@ -27,20 +55,41 @@ interface IBorrowService {
...
+     * @param currency The address of the currency of the loan
+     * @return The amount of required collateral
+     */
     function calculateRequiredCollateral(uint256 amount, address currency)
         external
         view
         returns (uint256);
 
+    /**
+     * @notice Gets the balance of the currency add as collateral
```
Comment (dvrosalesm) [inline contracts/services/IBorrowService.sol:91]:
```suggestion
     * @notice Gets the balance of the currency as collateral
```

File: contracts/services/ILendingService.sol [hunk 4]
```diff
@@ -4,20 +4,43 @@ pragma solidity ^0.8.16;
...
+     */
     event Lend(
         uint256 indexed listingId,
         address indexed lender,
         address indexed currency,
         uint256 amount
     );
 
+    /**
+     * @notice Lends funds to an user using a service listing
```
Comment (dvrosalesm) [inline contracts/services/ILendingService.sol:28]:
```suggestion
     * @notice Lends funds to a user from a service listing
```

File: contracts/services/IService.sol [hunk 5]
```diff
@@ -4,47 +4,125 @@ pragma solidity ^0.8.16;
...
+     */
     error FailedOperation(bytes data);
+    /**
+     * @notice Throws when the listing is not enabled
+     * @param listingId The id of the listing
+     */
     error ListingDisabled(uint256 listingId);
 
+    /**
+     * @notice Add a listing to the service
```
Comment (dvrosalesm) [inline contracts/services/IService.sol:51]:
```suggestion
     * @dev The listing id from the listing struct will be ignored and overwritten with an incremental id generated by the function
     * @notice Add a listing to the service
```

File: contracts/services/IService.sol [hunk 6]
```diff
@@ -4,47 +4,125 @@ pragma solidity ^0.8.16;
...
         returns (ServiceListing memory);
 
+    /**
+     * @notice Returns the number of listings
+     * @return count The number of listings
+     */
     function getListingsCount() external view returns (uint256);
 
+    /**
+     * @notice Allows service owner to update a giving listing
```
Comment (dvrosalesm) [inline contracts/services/IService.sol:82]:
```suggestion
     * @notice Allows service owner to update a given listing
```

File: contracts/services/IService.sol [hunk 7]
```diff
@@ -4,47 +4,125 @@ pragma solidity ^0.8.16;
...
+     * @param listing The listing to be updated
+     */
     function updateListing(ServiceListing memory listing) external;
 
+    /**
+     * @notice Returns balance on a given currency for the user on the service
```
Comment (dvrosalesm) [inline contracts/services/IService.sol:88]:
```suggestion
     * @notice Returns the balance of a given currency for the user on the service
```

File: contracts/services/Service.sol [hunk 8]
```diff
@@ -89,10 +126,19 @@ abstract contract Service is Ownable, IService {
         _addLiquidityInternal(amount, listingId);
     }
 
+    /**
+     * @notice Adds the given amount of liquidity from a given listing id
```
Comment (dvrosalesm) [inline contracts/services/Service.sol:130]:
```suggestion
     * @notice Adds the given amount of liquidity to a given listing id
```

**Review by dvrosalesm — APPROVED** [review]
Just left minor improvements, great work!

**Pull request merged**